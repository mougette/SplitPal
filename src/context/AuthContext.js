import React, { useState} from "react";
import createDataContext from './createDataContext';
import {Post} from '../components/RestPost';
import {Get} from '../components/RestGet';

const authReducer = (state, action) => {
  switch (action.type) {
    case 'signout':
      return {token: null, email: ''};
    case 'signin':
    return {
            email: action.payload.email,
            token: action.payload.token,
          };
    case 'signup':
      return {
        token: action.payload.token,
        email: action.payload.email,
      };
  }
};

const signup = dispatch => {
    const [account, setAccount] = useState("");
  return ({email, password, password2, firstName, lastName, phoneNumber}) => {
    Post("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/login",
                JSON.stringify({
                 email: email,
                 password: password,
                 password2: password2,
                    firstName: firstName,
                    lastName: lastName,
                    phoneNumber: phoneNumber,
                 }))
        .then(response => {
    console.log("Account: " + account + " Account End");
      if(response == "Account Successfully Made") {
              dispatch({
                  type: 'signup',
                  payload: {
                      token: 'some access token here',
                      email: email,
                  },
              });
      }
      else {
              alert(response)
      }
      }
      );


  };
};

const signin = dispatch => {
  return ({email, password}) => {
    // Do some API Request here
   Get("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/login","?email="+email+"&password="+password)
   .then(response =>{
    if(response == "User info correct"){
    dispatch({
      type: 'signin',
      payload: {
        token: 'some access token here',
        email: email,
      },
    });
    }});
  };
};

const signout = dispatch => {
  return () => {
    dispatch({type: 'signout'});
  };
};

export const {Provider, Context} = createDataContext(
  authReducer,
  {signin, signout, signup},
  {token: null, email: ''},
);
