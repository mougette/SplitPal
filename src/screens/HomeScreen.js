import React, {useState, useEffect, useContext} from 'react';
import {View, Text, StyleSheet, Image, TouchableOpacity, FlatList, SafeAreaView} from 'react-native';
import {Context as AuthContext} from '../context/AuthContext';
import SplitPalLogoComponent from '../components/SplitPalLogoComponent';
import Entry from "../components/EntryComponent";
import {Post} from '../components/RestPost';
import {Get} from '../components/RestGet';

const HomeScreen = ({navigation}) => {
  const {state} = useContext(AuthContext);
  const [DATA, setDATA] = useState("");
  const balance = DATA == "" ? DATA :DATA[0].TotalBalance
  const str = (balance > 0) ? "You are owed:" : "You owe:"
  const color = (balance > 0) ? "#2abb42" : "#bb2a2a"
  const isLoaded = navigation.isFocused()
  useEffect(() => {
  const unsubscribe = navigation.addListener('focus', () => {
  Get("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_2/balance","?userEmail="+state.email)
     .then(response => setDATA(response));
  });
  return unsubscribe
  },[navigation]);

  const renderItem = ( {item,index} ) => {
  console.log(item)
    return (
        <Entry
            image='https://reactnative.dev/img/tiny_logo.png'
            name={item.FirstName+" "+item.LastName}
            balance={item.Balance.toString()}
            onPress = {() => Post("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_2/transaction",
            JSON.stringify({sender: state.email,
                            groupID: "NULL",
                            transactions: [{item:"payment",
                                            amount:item.balance,
                                            usersOwed:item.Email}],}))}
        ></Entry>
    );
  };

  return (
    <View style={styles.master}>
      <SplitPalLogoComponent />
      <Text style={styles.header}>Grand Total</Text>

      <View style={styles.listItem}>
        <View style={{justifyContent: 'center', marginLeft: 25, flex:1}}>
          <Text style={{fontWeight:"bold"}}>{str}</Text>
        </View>
        <TouchableOpacity style={[styles.textButton, {backgroundColor: color}]}>
          <Text style={{color: "white", fontSize: 18}}>{DATA == "" ? DATA :DATA[0].TotalBalance}</Text>
        </TouchableOpacity>
      </View>

      <Text style={styles.header2}>Running Totals</Text>

      <SafeAreaView style={styles.containerTop}>
        <FlatList
            data={DATA == "" ? DATA : DATA.splice(1)}
            renderItem={renderItem}
            keyExtractor = {(item, index) => index.toString()}
            extraData={DATA}
        />
      </SafeAreaView>
    </View>
  );
};

const styles = StyleSheet.create({
  master: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#87a1b6',
  },
  header: {
    fontSize: 26,
    marginBottom: 10,
  },
  header2: {
    fontSize: 26,
    marginBottom: 0,
    marginTop: 10,
  },
  listItem:{
    backgroundColor: "#f5f5f5",
    padding: 10,
    marginVertical: 8,
    width:400,
    flexDirection:"row",
    borderRadius: 7,
    alignItems: "center",
    justifyContent: "center",
  },
  textButton: {
    fontSize: 20,
    borderWidth: 1,
    borderRadius: 7,
    overflow: "hidden",
    padding: 10,
    marginLeft: 10,
    width: '30%',
    color: '#2196F3',
    backgroundColor: '#2abb42',
    alignItems: "center",
  },
  containerTop: {
    flex: 2,
    marginTop: 15,
  },
});

export default HomeScreen;
