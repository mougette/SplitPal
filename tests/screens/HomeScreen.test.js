import React, {useContext} from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react-native'
import {Provider as AuthProvider} from '../../src/context/AuthContext.js';
import {Context as AuthContext} from '../../src/context/AuthContext';
import {navigate} from '@reach/router'
import HomeScreen from '../../src/screens/HomeScreen'
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';

//mock fetch
global.fetch = jest.fn().mockImplementation(() => {
      var p = new Promise((resolve, reject) => {
        resolve({
          ok: true,
          Id: '123',
          json: function() {
            return {Id: '123'}
          }
        });
      });

      return p;
  });


describe('Test Hooks', () => {
    const Stack = createBottomTabNavigator();
    it('navigates on button press', () => {
    const { getByText } = render(
    <AuthProvider>
    <NavigationContainer>
    <Stack.Navigator>
    <Stack.Screen name="HomeScreen" component={HomeScreen} />
    </Stack.Navigator>
    </NavigationContainer>
    </AuthProvider>);
    expect(1).toBe(1);
    });


});