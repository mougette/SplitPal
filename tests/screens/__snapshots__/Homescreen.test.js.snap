// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Hooks render 1`] = `undefined`;

exports[`Test Hooks render 2`] = `undefined`;
